sycl/plugins/esimd_cpu/pi_esimd_cpu.cpp

```diff
@@ -476,7 +479,7 @@ struct ESIMDDeviceInterfaceImpl : public ESIMDDeviceInterface {
   ESIMDDeviceInterfaceImpl() {
 #ifdef __GNUC__
     // Linux
-    void *handle = dlopen("libcm.so", RTLD_LAZY);
+    void *handle = dlopen("libcm.so", RTLD_LAZY | RTLD_NODELETE);
```

Comment:
Could you please clarify why RTLD_NODELETE is passed here?

Reply:
BTW. Is there any reason why libcm.so cannot be linked during compilation? It would help to avoid having to dlopen and dlsym?
Also the situation when dlopen or dlsym fail should be gracefully handled here.

Reply:
The '-lcm' option will be applied in `sycl/plugins/esimd_cpu/CMakeLists.txt` for toolchain build. This is invisible to customers.

However, for kernel compilation, we need either this 'dlsym/GetProcAddress' in plug-in initialization or '-lcm' as command line argument for clang++ in order to use intrinsic implementations provided by libcm.

I'll add some more details for 'dlopen/dlsym' failure - as well as 'RTDL_NODELTE'.